crates/rabbitmq/src/producer.rs

```
@@ -42,6 +42,21 @@ where
         &self,
         val: impl std::borrow::Borrow<Q::Message>,
         routing_key: Option<&str>,
+    ) -> Result<()> {
+        let _ = &self.write_with_props(val, routing_key, None).await;
```

ouch. you're swallowing any errors.

---

can just be 1 line, no semicolon
`&self.write_with_props(val, routing_key, None)`
and remove async from function

---

oof okay. nice catch!